packages/preview/lemmify-0.1.0/README.md

```diff
@@ -0,0 +1,274 @@
+to typst (for example if user-defined element functions
+are introduced). But no functionality should be lost.
+
+## Basic Usage
+
+To get started with Lemmify, follow these steps:
+
+1. Import the Lemmify library:
+```typst
+#import "@preview/lemmify:0.1.0": *
+```
+
+2. Define the default styling for a few default theorem types:
+```typst
+#let (
+  theorem, lemma, corollary,
+  remark, proposition, example,
+  proof, rules: thm-rules
+) = default-theorems("thm-group", lang: "en")
+```
+
+3. Apply the generated styling:
+```typst
+#show: thm-rules
+```
+
+4. Create theorems, lemmas, and proofs using the defined styling:
+```typst
+#theorem(name: "Some theorem")[
+  Theorem content goes here.
+]<thm>
+
+#proof[
+  Complicated proof.
+]<proof>
+
+@proof @thm[Some thoerem]
+```
+
+5. Customize the styling further using show rules. For example, to add a red box around proofs:
+```
+#show thm-selector("thm-group", subgroup: "proof"): it => box(
+  it,
+  stroke: red + 1pt,
+  inset: 1em
+)
+```
+
+The result should now look something like this:
+
+![image](https://github.com/Marmare314/lemmify/assets/49279081/090381b2-9906-4b53-ae6c-5382840265de)
+
+
+## Useful examples
+
+If you do not want to reset the theorem counter on headings
+you can use the `max-reset-level` parameter:
+
+```typst
+default-theorems("thm-group", max-reset-level: 0)
+```
+
+It specifies whats the highest level at which the counter is reset. To manually reset the counter you can use the
+`thm-reset-counter` function.
+
+---
+
+By specifying `numbering: none` you can create unnumbered
+theorems.
+
+```typst
+#example(numbering: none)[
+  Some example.
+]
+```
+
+To make all examples unnumbered you could use the following code:
+
+```typst
+#let example = example.with(numbering: none)
+```
+
+---
+
+To create other types (or subgroups) of theorems you can use the
+`new-theorems` function.
+
+```typst
+#let (note, rules) = new-theorems("thm-group", ("note": text(red)[Note]))
+#show: rules
+```
+
+If you have already defined custom styling you will notice that
+the newly created theorem does not use it.
+You can create a dictionary to make applying it again easier.
+
+```typst
+#let my-styling = (
+  thm-styling: thm-styling-simple,
+  thm-numbering: ...,
+  ref-styling: ...
+)
+
+#let (note, rules) = new-theorems("thm-group", ("note": "Note), ..my-styling)
+```
+
+---
+
+By varying the `group` parameter you can create independently numbered theorems:
+
+```typst
+#let (
+  theorem, proof,
+  rules: thm-rules-a
+) = default-theorems("thm-group-a")
+#let (
+  definition,
+  rules: thm-rules-b
+) = default-theorems("thm-group-b")
+
+#show: thm-rules-a
+#show: thm-rules-b
+```
+
+## Example
+
+```typst
+#import "@preview/lemmify:0.1.0": *
+
+#let my-thm-style(
+  thm-type, name, number, body
+) = block[#{
+  grid(columns: (1fr, 3fr), column-gutter: 1em)[#{
+    stack(spacing: .5em, strong(thm-type), number, emph(name))
+  }][#body]
+}]
```

Comment:
This could be simplified a lot, down to this:
```
#let my-thm-style(
  thm-type, name, number, body
) = grid(
  columns: (1fr, 3fr),
  column-gutter: 1em,
  stack(spacing: .5em, strong(thm-type), number, emph(name)),
  body,
)
```
That's not a necessarily a blocker for merging, but I just wanted to note this in case you want to update it.

Reply:
Ah thanks, that looks a lot better!